src/plugin/debugger/mhpdbgc.c

```diff
@@ -1077,10 +1067,26 @@ static const char *get_name_from_mcb(struct MCB *mcb, int *is_lnk)
     }
     return dos;
   }
-  snprintf(name, sizeof name, "%s", mcb->name);
-  if (!is_valid_program_name(name))
-    snprintf(name, sizeof name, "%05d", mcb->owner_psp);
 
+  // Are we the progenitor?
+  p1 = mcb;
+  p2 = MK_FP32(p1->owner_psp - 1, 0);
+  if (p1 == p2) {
+    strlcpy(name, p1->name, 8 + 1); // Source not null terminated if 8 chars
+    return name;
+  }
+
+  // Are we 1st decendant?
+  p1 = MK_FP32(mcb->owner_psp - 1, 0);
+  p2 = MK_FP32(p1->owner_psp - 1, 0);
+  if (p1 == p2) {
```

Comment:
It's checking if the `parent` == `parent's parent` if it is then use the parent name.

Reply:
Ah, "then use parent's name" is what
I missed.

@ecm-pushbx OK to apply?

Reply:
I thought about trying to detect if an MCB's data is an environment block like QEMM's MFT does, but I can't think of a very good heuristic.

Reply:
You need to search the region
for "\0\1\0" and see that there is
1 ASCIIZ string after that marker,
and at least 1 - before.

Reply:
Thanks, I'll give it a try.

Reply:
You can detect the environment by checking that the owner is a self-owned MCB with a valid PSP (first word = 20CDh) and that the environment segment reference in the PSP matches the memory block's segment.

Reply:
I would add the explicit check at least for owner = 0 and owner = 8. Else if the IVT memory paragraph at 0:70h contains an apparent "owner" field of 8 then invalid data will be read from the IVT.

Reply:
> 20CDh) and that the environment segment reference in the PSP matches the memory block's segment.

Good idea. :)
Andrew asked for the heuristic, and
I guess I interpreted his question too literally
and gave a heuristic, instead of a proper check.